src/form-fields/file-upload.js

```diff
@@ -112,6 +113,15 @@ class FileUpload extends React.Component {
     }
 
     onUpload = async e => {
+        // TODO: Remove once React is updated to version 17. In React 16 and
+        // earlier, SyntheticEvent objects are pooled for performance reasons -
+        // event properties are set to null once the event handler returns
+        // unless event.persist is called(). This optimisation was removed in
+        // React 17 as it does not improve performance on modern browsers. See
+        // https://reactjs.org/docs/legacy-event-pooling.html and
+        // https://reactjs.org/blog/2020/08/10/react-v17-rc.html#no-event-pooling.
+        e.persist()
```

Comment:
You could create a reference to the files but I wanted the fix to be explicit so that 1) it wouldn't disappear during a refactor for what would appear to be equivalent code, and 2) it could be removed as soon as we upgrade the react version in the platform to `v17`.

The second time that `e.target.files` is read is after this `await` and so the event object is accessed after the (`async`) event handler returns:
https://github.com/dhis2/settings-app/blob/7900da99c5417a6e7702826340c06b2d6ed19225/src/form-fields/file-upload.js#L134

Reply:
> The second time that `e.target.files` is read is after this await 

Thanks for clarifying

> You could create a reference to the files but I wanted the fix to be explicit

IMO persisting the event because you want to hang on to the files isn't that explicit. I'd argue that storing the files in a const is more explicit, and keeping that const around even after we upgrade to React v17 is a good thing, because it makes the code more readable.